pymc3/step_methods/slicer.py

```diff
@@ -44,33 +44,39 @@ def __init__(self, vars=None, w=1., tune=True, model=None, **kwargs):
         super(Slice, self).__init__(vars, [self.model.fastlogp], **kwargs)
 
     def astep(self, q0, logp):
-        self.w = np.resize(self.w, len(q0))
-        y = logp(q0) - nr.standard_exponential()
-
-        # Stepping out procedure
-        q_left = q0 - nr.uniform(0, self.w)
-        q_right = q_left + self.w
-
-        while (y < logp(q_left)).all():
-            q_left -= self.w
-
-        while (y < logp(q_right)).all():
-            q_right += self.w
-
-        q = nr.uniform(q_left, q_right, size=q_left.size)  # new variable to avoid copies
-        while logp(q) <= y:
-            # Sample uniformly from slice
-            if (q > q0).all():
-                q_right = q
-            elif (q < q0).all():
-                q_left = q
-            q = nr.uniform(q_left, q_right, size=q_left.size)
-
+        self.w = np.resize(self.w, len(q0))  # this is a repmat
+        q = np.copy(q0)  # TODO: find out if we need this
+        ql = np.copy(q0)  # l for left boundary
+        qr = np.copy(q0)  # r for right boudary
+        for i in range(len(q0)):
```

Comment:
Feel free to ignore, but this loop might be cleaner using

`for i, (point, width) in enumerate(zip(q, self.w)):`